core/src/main/java/com/linecorp/armeria/client/AbstractEventLoopState.java

```diff
@@ -29,6 +29,8 @@ static AbstractEventLoopState of(List<EventLoop> eventLoops, int maxNumEventLoop
                                      DefaultEventLoopScheduler scheduler) {
         if (maxNumEventLoops == 1) {
             return new OneEventLoopState(eventLoops, scheduler);
+        } else if (maxNumEventLoops <= 4) {
```

Comment:
Should we now bump this number?

```suggestion
        } else if (maxNumEventLoops <= 128) {
```

Reply:
Here is the result of the benchmark:
```
Benchmark                                (arrayBased)  (maxNumEventLoops)   Mode  Cnt       Score       Error  Units
EventLoopStateBenchmark.firstInFirstOut          true                  32  thrpt    5  805478.672 ± 24188.976  ops/s
EventLoopStateBenchmark.firstInFirstOut          true                  64  thrpt    5  267960.971 ±  4286.936  ops/s
EventLoopStateBenchmark.firstInFirstOut          true                 128  thrpt    5   76843.522 ±   390.589  ops/s
EventLoopStateBenchmark.firstInFirstOut          true                 256  thrpt    5   20250.525 ±    29.497  ops/s
EventLoopStateBenchmark.firstInFirstOut         false                  32  thrpt    5  351819.729 ±  1081.003  ops/s
EventLoopStateBenchmark.firstInFirstOut         false                  64  thrpt    5  147116.804 ±  1008.900  ops/s
EventLoopStateBenchmark.firstInFirstOut         false                 128  thrpt    5   60397.639 ±   305.486  ops/s
EventLoopStateBenchmark.firstInFirstOut         false                 256  thrpt    5   25557.867 ±   109.960  ops/s

EventLoopStateBenchmark.lastInFirstOut           true                  32  thrpt    5  806221.347 ±  2136.376  ops/s
EventLoopStateBenchmark.lastInFirstOut           true                  64  thrpt    5  273869.470 ±  1218.701  ops/s
EventLoopStateBenchmark.lastInFirstOut           true                 128  thrpt    5   76529.580 ±  1370.584  ops/s
EventLoopStateBenchmark.lastInFirstOut           true                 256  thrpt    5   20265.238 ±    30.240  ops/s
EventLoopStateBenchmark.lastInFirstOut          false                  32  thrpt    5  344827.240 ± 19401.810  ops/s
EventLoopStateBenchmark.lastInFirstOut          false                  64  thrpt    5  143315.611 ±  2398.936  ops/s
EventLoopStateBenchmark.lastInFirstOut          false                 128  thrpt    5   60153.871 ±   110.086  ops/s
EventLoopStateBenchmark.lastInFirstOut          false                 256  thrpt    5   25732.990 ±   440.076  ops/s
```

Reply:
I have found out that there's overhead of `ArrayList` type casting.
![flame-graph-cpu](https://user-images.githubusercontent.com/25103250/224714437-f0e70f92-83be-4405-882b-2187123ba272.svg)

Reply:
This is the version that uses raw array instead of array list:
```
Benchmark                                (arrayBased)  (maxNumEventLoops)   Mode  Cnt        Score   Error  Units
EventLoopStateBenchmark.firstInFirstOut          true                  32  thrpt    2  1050623.524          ops/s
EventLoopStateBenchmark.firstInFirstOut          true                  64  thrpt    2   382256.852          ops/s
EventLoopStateBenchmark.firstInFirstOut          true                 128  thrpt    2   121629.625          ops/s
EventLoopStateBenchmark.firstInFirstOut          true                 256  thrpt    2    33119.314          ops/s
EventLoopStateBenchmark.firstInFirstOut         false                  32  thrpt    2   598772.414          ops/s
EventLoopStateBenchmark.firstInFirstOut         false                  64  thrpt    2   240048.384          ops/s
EventLoopStateBenchmark.firstInFirstOut         false                 128  thrpt    2   101510.138          ops/s
EventLoopStateBenchmark.firstInFirstOut         false                 256  thrpt    2    43237.621          ops/s

EventLoopStateBenchmark.lastInFirstOut           true                  32  thrpt    2  1060206.127          ops/s
EventLoopStateBenchmark.lastInFirstOut           true                  64  thrpt    2   382819.196          ops/s
EventLoopStateBenchmark.lastInFirstOut           true                 128  thrpt    2   121602.774          ops/s
EventLoopStateBenchmark.lastInFirstOut           true                 256  thrpt    2    33603.237          ops/s
EventLoopStateBenchmark.lastInFirstOut          false                  32  thrpt    2   657403.575          ops/s
EventLoopStateBenchmark.lastInFirstOut          false                  64  thrpt    2   285346.029          ops/s
EventLoopStateBenchmark.lastInFirstOut          false                 128  thrpt    2   122770.175          ops/s
EventLoopStateBenchmark.lastInFirstOut          false                 256  thrpt    2    53913.492          ops/s
```